### ISSUE TITLE
Replace usage of os.path.walk in favor of twisted.python.filepath.FilePath.walk

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5139]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5139|
|Type|task|
|Created|2011-06-06 17:48:18Z|
|Branch|https://github.com/twisted/twisted/tree/test-doc-cleanup-5139|

Twisted emits the following py3k warning:

```
In 3.x, os.path.walk is removed in favor of os.walk.
```

For the following files:

```
./twisted/lore/process.py:33:        os.path.walk(topdir, self.walk, prefix)
./twisted/scripts/tkunzip.py:174:    os.path.walk(path, justlist, all)
./twisted/scripts/tkunzip.py:200:    os.path.walk(path, countPys, countl)
./twisted/test/test_doc.py:26:        os.path.walk(os.path.dirname(twisted.__file__),
```

`twisted.lore` is being phased out so that one can be ignored.

Attachments:

* [test_doc.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/0f5/0f5e083c0f34103b244e88e5efd8d124c35e8267/20b5cdf9ae0a10a299603a070688dc4f5d79af3f.patch) (2946 bytes) - added by thijs on 2011-10-07 17:12:29Z - 
* [test_doc2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/0f5/0f5e083c0f34103b244e88e5efd8d124c35e8267/a26866f5bc978013fc3d42c355c2316d3cd852a3.patch) (2557 bytes) - added by thijs on 2011-10-12 12:35:28Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5139 5139
type__task task
reporter__thijs thijs
priority__low low
milestone__Python_3_x Python-3.x
branch__branches_test_doc_cleanup_5139 branches/test-doc-cleanup-5139
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1307382498000000 1307382498000000
changetime__1318868537000000 1318868537000000
version__None None
owner__thijs thijs
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

We don't want or need `twisted/scripts/tkunzip.py`, it should be deprecated any then removed.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_3):
> We don't want or need `twisted/scripts/tkunzip.py`, it should be deprecated any then removed.

Opened [#5140](https://github.com/twisted/twisted/issues/5140) for that.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

This is an okay approach.  The new code generates the same results as the old code.  For some reason it's about 10x slower though.  It still only takes 0.5 seconds here, but that's up from 0.05 seconds in trunk.  Not a huge loss, but it would be nice to avoid.

Apart from that, what would be really special is to just use `twisted.python.modules` to replace all of this custom half-baked package and module discovery code.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra removed owner|
|-|-|

Replaced the `walk()` stuff with `t.p.modules` in the new patch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Cool, thanks.  Looks good.  Maybe a good idea to check into a branch and get buildbot to run before committing to trunk, since this touches lots of Twisted and who knows about crazy platform-specific issues.  Apart from any issues with that, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_9"></a>|@thijstriemstra commented|
|-|-|

(In [32896]) Branching to 'test-doc-cleanup-5139'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_10"></a>|@thijstriemstra commented|
|-|-|

(In [32897]) Apply test_doc2.patch, refs [#5139](https://github.com/twisted/twisted/issues/5139)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_11"></a>|@thijstriemstra set status to `assigned`|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/test-doc-cleanup-5139).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [32904]) Merge test-doc-cleanup-5139: Replace the custom package and module discovery code with t.p.modules.getModule. This gets rid of a Python3k deprecation warning for os.path.walk.

Author: thijs
Reviewer: exarkun
Fixes: [#5139](https://github.com/twisted/twisted/issues/5139)
